src/main/java/com/iota/iri/service/validation/impl/TransactionSolidifierImpl.java

```diff
@@ -0,0 +1,389 @@
+        if(maxProcessedTransactions != Integer.MAX_VALUE) {
+            maxProcessedTransactions += analyzedHashes.size();
+        }
+        boolean solid = true;
+        final Queue<Hash> nonAnalyzedTransactions = new LinkedList<>(Collections.singleton(hash));
+        Hash hashPointer;
+        while ((hashPointer = nonAnalyzedTransactions.poll()) != null) {
+            if (!analyzedHashes.add(hashPointer)) {
+                continue;
+            }
+
+            if (analyzedHashes.size() >= maxProcessedTransactions) {
+                return false;
+            }
+
+            TransactionViewModel transaction = fromHash(tangle, hashPointer);
+            if (isUnsolidWithoutEntryPoint(transaction, hashPointer)) {
+                if (transaction.getType() == PREFILLED_SLOT) {
+                    solid = false;
+                    checkRequester(hashPointer);
+                } else {
+                    nonAnalyzedTransactions.offer(transaction.getTrunkTransactionHash());
+                    nonAnalyzedTransactions.offer(transaction.getBranchTransactionHash());
+                }
+            }
+        }
+        if (solid) {
+          updateTransactions(analyzedHashes);
+        }
+        analyzedHashes.clear();
+        return  solid;
+    }
+
+
+    /**
+     * Check if a transaction is present in the {@link #transactionRequester}, if not, it is added.
+     * @param hashPointer   The hash of the transaction to request
+     */
+    private void checkRequester(Hash hashPointer){
+        if (!transactionRequester.isTransactionRequested(hashPointer)) {
+            transactionRequester.requestTransaction(hashPointer);
+        }
+    }
+
+    /**
+     * Iterate through analyzed hashes and place them in the {@link #transactionsToBroadcast} queue
+     * @param hashes    Analyzed hashes from the {@link #checkSolidity(Hash)} call
+     */
+    private void updateTransactions(Set<Hash> hashes) {
+        hashes.forEach(hash -> {
+            try {
+                TransactionViewModel tvm = fromHash(tangle, hash);
+                tvm.updateHeights(tangle, snapshotProvider.getInitialSnapshot());
+
+                if(!tvm.isSolid()){
+                    tvm.updateSolid(true);
+                    tvm.update(tangle, snapshotProvider.getInitialSnapshot(), "solid|height");
+                }
+                addToBroadcastQueue(tvm);
+                addToPropagationQueue(tvm.getHash());
+            } catch (Exception e) {
+                log.info(e.getMessage());
+            }
+        });
+    }
+
+    /**
+     * Returns true if transaction is not solid and there are no solid entry points from the initial snapshot.
+     */
+    private boolean isUnsolidWithoutEntryPoint(TransactionViewModel transaction, Hash hashPointer) throws Exception{
+        if(!transaction.isSolid()){
+            if(!snapshotProvider.getInitialSnapshot().hasSolidEntryPoint(hashPointer)){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4339340)